src/app/public/modules/auth-http-client/auth-context-provider.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  Injectable
+} from '@angular/core';
+
+import {
+  BBContextArgs
+} from '@blackbaud/auth-client';
+
+import {
+  BBAuthClientFactory
+} from '@skyux/auth-client-factory';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SkyAuthContextProvider {
+
+  public ensureContext(args: BBContextArgs) {
```

Comment:
Since the goal of this service is to abstract away the `BBAuth` API, I would avoid exposing any of its API to consumers of this service, which includes the use of `BBContextArgs` as a parameter type. In `SkyAuthTokenProvider`, for instance, we created new types to expose to the consumer. As for the type definition, it simply inherits the type from `BBAuth`.

https://github.com/blackbaud/skyux-http/blob/master/src/app/public/modules/auth-http-client/auth-token-provider.ts#L47
https://github.com/blackbaud/skyux-http/blob/master/src/app/public/modules/auth-http-client/auth-get-token-args.ts

Reply:
Good idea.